packages/client/src/utils/fetch/QueryFetch.js

```diff
@@ -16,7 +16,10 @@ export default class QueryFetch extends DataFetch {
     if (!datasource?._id) {
       return null
     }
-    return await fetchQueryDefinition(datasource._id)
+    const definition = await fetchQueryDefinition(datasource._id)
+    // After getting the definition of query, it loses "fields" attribute because of security reason from the server. However, this attribute needs to be inside of defintion for pagination.
+    definition.fields = datasource.fields
```

Comment:
Could we first check if `fields` exist? In dev apps the fields will be present inside `definition`, and will be up to date. The ones stored here in `datasource` are potentially out of date, as the value of the datasource is saved at the moment you select it in the builder settings.

So we should just change this to:
```js
if (!definition.fields) {
  definition.fields = datasource.fields
}
```

Reply:
I've changed it. Thanks for your feedback. It helps me to understand the logic.